feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/domain/validation/SwapValidations.kt

```diff
@@ -0,0 +1,104 @@
+typealias SwapValidation = Validation<SwapValidationPayload, SwapValidationFailure>
+typealias SwapValidationSystemBuilder = ValidationSystemBuilder<SwapValidationPayload, SwapValidationFailure>
+
+fun SwapValidationSystemBuilder.availableSlippage(assetExchange: AssetExchange) = validate(
+    SwapSlippageRangeValidation(
+        assetExchange = assetExchange
+    )
+)
+
+fun SwapValidationSystemBuilder.swapFeeSufficientBalance() = validate(
+    SwapFeeSufficientBalanceValidation()
+)
+
+fun SwapValidationSystemBuilder.swapSmallRemainingBalance(
+    assetSourceRegistry: AssetSourceRegistry,
+    chainRegistry: ChainRegistry
+) = validate(
+    SwapSmallRemainingBalanceValidation(
+        assetSourceRegistry,
+        chainRegistry
+    )
+)
+
+fun SwapValidationSystemBuilder.rateNotExceedSlippage(sharedQuoteValidationRetriever: SharedQuoteValidationRetriever) = validate(
+    SwapRateChangesValidation { sharedQuoteValidationRetriever.retrieveQuote(it).getOrThrow().swapRate() }
+)
+
+fun SwapValidationSystemBuilder.enoughLiquidity(sharedQuoteValidationRetriever: SharedQuoteValidationRetriever) = validate(
+    SwapEnoughLiquidityValidation { sharedQuoteValidationRetriever.retrieveQuote(it) }
+)
+
+fun SwapValidationSystemBuilder.sufficientBalanceInFeeAsset() = sufficientBalance(
+    available = { it.feeAsset.transferable },
+    amount = { BigDecimal.ZERO },
+    fee = { it.feeAsset.token.amountFromPlanks(it.swapFee.networkFee.amount) },
+    error = { payload, availableToPayFees ->
+        SwapValidationFailure.NotEnoughFunds.InCommissionAsset(
+            chainAsset = payload.feeAsset.token.configuration,
+            fee = payload.feeAsset.token.amountFromPlanks(payload.swapFee.networkFee.amount),
+            availableToPayFees = availableToPayFees
+        )
+    }
+)
+
+fun SwapValidationSystemBuilder.sufficientBalanceInUsedAsset() = sufficientBalance(
+    available = { it.detailedAssetIn.asset.transferable },
+    amount = { it.detailedAssetIn.amount },
+    fee = { BigDecimal.ZERO },
+    error = { _, _ ->
+        SwapValidationFailure.NotEnoughFunds.InUsedAsset
+    }
+)
+
+// TODO change to DeadRecipientValidation
+fun SwapValidationSystemBuilder.sufficientRecipientBalanceToStayAboveED(
+    enoughTotalToStayAboveEDValidationFactory: EnoughTotalToStayAboveEDValidationFactory
+) {
+    enoughTotalToStayAboveEDValidationFactory.validate(
+        fee = { BigDecimal.ZERO },
+        total = { it.detailedAssetOut.asset.total + it.detailedAssetOut.amount },
+        chainWithAsset = { ChainWithAsset(it.detailedAssetOut.chain, it.detailedAssetOut.asset.token.configuration) },
+        error = { payload, existentialDeposit ->
+            SwapValidationFailure.AmountOutIsTooLowToStayAboveED(
+                payload.detailedAssetOut.asset.token.configuration,
+                payload.detailedAssetOut.amount,
+                existentialDeposit
+            )
+        }
+    )
+}
```

Comment:
What do you think about using this validation EnoughTotalToStayAboveEDValidation for this more specific case?

Reply:
While the usage semantics is a little bit off, I agree that it works. I would only suggest to rename `sufficientRecipientBalanceToStayAboveED` to `sufficientAssetOutBalanceToStayAboveED`

Reply:
Agree. Fixed it for nice semantic